src/java_bytecode/java_object_factory.cpp

```diff
@@ -151,7 +151,7 @@ class java_object_factoryt
 /// Generates code for allocating a dynamic object. This is used in
 /// allocate_object and for allocating strings in the library preprocessing.
 /// \param target_expr: expression to which the necessary memory will be
-///   allocated
+///   allocated, its type should be pointer to `allocate_type`
```

Comment:
I think  you are talking about `allocate_dynamic_object_with_decl` , but this one still has `allocate_type` as an argument